pkg/operations/cordondrainvm.go

```diff
@@ -173,24 +174,31 @@ func (o *drainOperation) deleteOrEvictPods(pods []v1.Pod) error {
 }
 
 func (o *drainOperation) evictPods(pods []v1.Pod, policyGroupVersion string) error {
+	ctx, cancel := context.WithCancel(context.Background())
 	doneCh := make(chan bool, len(pods))
 	errCh := make(chan error, 1)
 
 	for _, pod := range pods {
-		go func(pod v1.Pod, doneCh chan bool, errCh chan error) {
+		go func(ctx context.Context, pod v1.Pod, doneCh chan bool, errCh chan error) {
 			var err error
+		doneEviction:
 			for {
-				err = o.client.EvictPod(&pod, policyGroupVersion)
-				if err == nil {
-					break
-				} else if apierrors.IsNotFound(err) {
-					doneCh <- true
-					return
-				} else if apierrors.IsTooManyRequests(err) {
-					time.Sleep(5 * time.Second)
-				} else {
-					errCh <- errors.Wrapf(err, "error when evicting pod %q", pod.Name)
+				select {
+				case <-ctx.Done():
 					return
+				default:
+					err = o.client.EvictPod(&pod, policyGroupVersion)
+					if err == nil {
+						break doneEviction
+					} else if apierrors.IsNotFound(err) {
+						doneCh <- true
+						return
+					} else if apierrors.IsTooManyRequests(err) {
+						time.Sleep(5 * time.Second)
+					} else {
+						errCh <- errors.Wrapf(err, "error when evicting pod %q", pod.Name)
+						return
+					}
 				}
 			}
 			podArray := []v1.Pod{pod}
```

Comment:
I don't see how this block of code will ever be reached. When we do `break doneEviction` above we will return from the go func entirely.

Reply:
`break doneEviction` will only break the 'for' loop when the err is nil. If the err is not nil, the block has a chance to be reached.

Reply:
And `break doneEviction` will not return from the go func entirely. `_, err = o.client.WaitForDelete(o.logger, podArray, true)` will be reached to wait for Pod Deletion.